### DISCUSSION TITLE
Why prisma use too much memory

### DISCUSSION THREAD

**Discussion by longdhit** [prisma/prisma#22386] [Q&A]
### Question

When i use prisma i see it spend too much memory ? Do anyone know how to reduce it ?
My table have 4908236 record
Stable memory
![image](https://github.com/prisma/prisma/assets/23386573/94173519-a238-4726-aa04-ebcf1c48022e)
With Prisma 
![prisma](https://github.com/prisma/prisma/assets/23386573/3680f485-257a-454b-b578-af32b921d88e)
With Prisma raw query
![prisma2](https://github.com/prisma/prisma/assets/23386573/46b33743-eb35-4a24-b7e8-67ce6ddaa4d1)
With mysql2
![image](https://github.com/prisma/prisma/assets/23386573/c8e21715-03ad-4a86-bfcb-fc73e26ef465)

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Windows
- Database: MySQL
- Node.js version: v18.16.1

- prisma  : 4.16.2
- @prisma/client  : 4.14.1

**Comment by ludralph** [comment]
Hi @longdhit 👋 

Thank you for raising this question.

I can see you are using Prisma version 4.16.2 and `@prisma/client` 4.14.1. You should note that we have had some performance improvements since Prisma version 5 as can be seen in this blog [post](https://www.prisma.io/blog/prisma-5-f66prwkjx72s). The latest version is Prisma version [5.7.0 ](https://github.com/prisma/prisma/releases/tag/5.7.0). I will advise you update to Prisma version 5 and see if performance improves for you. Also, always ensure you are using the current version for both the dependencies Prisma and `@prisma/client`. To get more insights on how Prisma interacts with your database, you can enable [tracing](https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/opentelemetry-tracing)

**Reply by longdhit** [reply]
Hi @ludralph  i have create issue report this bug in here https://github.com/prisma/prisma/issues/22387
I tried to use prisma version 5.7.0 but it still got problem. It seem it bug when using prisma because data about 5 million records about ~3GB memory but i got more than 10GB
